### PR TITLE
[PATCH API-NEXT v2] Set error flag on crypto errors

### PR DESCRIPTION
**Pull request by lumag**

### CODE REVIEW

File: test/validation/api/crypto/odp_crypto_test_inp.c [hunk 1]
```diff
@@ -270,6 +273,11 @@ static int alg_packet_op_enq(odp_packet_t pkt,
 		return rc;
 	}
 
+	if (!result.ok)
+		CU_ASSERT(odp_packet_has_error(pkt))
+	else
+		CU_ASSERT(!odp_packet_has_error(pkt));
+
```
Comment (Bill-Fischofer-Linaro) [inline test/validation/api/crypto/odp_crypto_test_inp.c]: `CU_ASSERT()` needs to be enclosed in braces when used in an `if/else` statement. However a simpler test is just:
```
CU_ASSERT(result.ok != odp_packet_has_error(pkt));
```

File: test/validation/api/crypto/odp_crypto_test_inp.c [hunk 2]
```diff
@@ -201,6 +201,9 @@ static int alg_packet_op(odp_packet_t pkt,
 		return rc;
 	}
 
+	if (!result.ok)
+		CU_ASSERT(odp_packet_has_error(pkt));
+
```
Comment (Bill-Fischofer-Linaro) [inline test/validation/api/crypto/odp_crypto_test_inp.c]: `CU_ASSERT(result.ok || odp_packet_has_error(pkt));` is simpler.

File: test/validation/api/crypto/odp_crypto_test_inp.c (hunk 1 above)
Comment (lumag) [reply test/validation/api/crypto/odp_crypto_test_inp.c]: ok.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**